modules/cli/src/main/scala/scala/cli/commands/SetupIde.scala

```diff
@@ -148,14 +149,22 @@ object SetupIde extends ScalaCommand[SetupIdeOptions] {
     val json                      = gson.toJson(details)
     val scalaCliOptionsForBspJson = write(options.shared)
 
+    if (bspJsonDestination.toString.contains(".cache"))
```

Comment:
We shouldn't detect that `.bsp` would land in the home dir this way… At a minimum, that check should have been something like `bspJsonDestination.segments.contains(".cache")`, but that would only work on Linux, and wouldn't always mean that Scala CLI decided on its own to put things there (users might have asked for it, and we should just let them in this case).

We should probably keep track of how `workspace` is computed in `Inputs` (from source paths / in home dir / forced by users). If this says the workspace was put in home dir by Scala CLI, and that workspace is used by `SharedBspFileOptions.bspDetails`, then we should print an error message.